includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -230,7 +230,7 @@ public static function extract_by_filename_or_filesystem( $dimensions ) {
 				$image_size = ( ! empty( $image_size ) && is_array( $image_size ) ) ? $image_size : [];
 			}
 
-			if ( ( empty( $image_size ) || ! is_array( $image_size ) ) && file_exists( $image_file ) ) {
+			if ( ( empty( $image_size ) ) && file_exists( $image_file ) ) {
```

Comment:
Redundant indeed since we check if it is an array above. Nevertheless, this can be further cleaned up a tiny bit more to remove the extra parentheses:

```suggestion
			if ( empty( $image_size ) && file_exists( $image_file ) ) {
```